src/frontend/utils/input-output.ts

```diff
@@ -0,0 +1,3 @@
+export interface InputOutput {
+  [key: string]: {alias: string};
```

Comment:
Now that I look at it, since you're using `hasOwnProperty` you can just make it a map directly to string instead of refcell of string.